packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -396,18 +396,28 @@ function buildBodyParameter(
     }
     return bodyParameter.optional
       ? `body: typeof ${bodyParameter.clientName} === 'string'
-    ? uint8ArrayToString(${bodyParameter.clientName}, "${
-          bodyParameter.type.format ?? "base64"
```

Comment:
You can see here, it passes bodyParameter.type.format but in here https://github.com/Azure/autorest.typescript/pull/2076/files#diff-06c7fcb35721db0a614bcf203b1b553efed5e0ac593c4a987e6055218361fc1dL911 the format is passed from outside.